lib/Model/device.dart

```diff
@@ -24,14 +25,20 @@ class Device {
   static Device fromJson(Map<String, dynamic> json) {
     var i = Device();
     i.guid = json['guid'].toString();
-    i.address = json['address'].toString();
+    if (json['address'].toString() != null) {
```

Comment:
not the  correct/elegant way for null comparison

Reply:
Use the `??=` operator instead?

`json['address'] ??= defaultIP;`